src/vision/status_quo/barbara_trims_a_stacktrace.md

```diff
@@ -0,0 +1,125 @@
+stack backtrace:
+   1: stacks_rs::process_one::{{closure}} at ./src/main.rs:16:5
+   3: stacks_rs::process_many::{{closure}} at ./src/main.rs:10:5
+   5: stacks_rs::main::{{closure}}::{{closure}} at ./src/main.rs:4:9
+   7: stacks_rs::main::{{closure}} at ./src/main.rs:3:5
+  13: <tokio stuff> 
+  19: stacks_rs::main
+note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
+```
+
+Based on this, she is able to figure out who to ping about the problem. She pastes her reduced stack trace into the issue pings Alan, who is responsible that module. Alan thanks her for reducing the stack trace and mentions, "Oh, when I used to work in C#, this is what the stack traces always looked out. I miss those days."
+
+Fin.
+
+## 🤔 Frequently Asked Questions
+
+*Here are some standard FAQ to get you started. Feel free to add more!*
+
+* **What are the morals of the story?**
+    * Rust stack traces -- but async stack traces in particular -- reveal lots of implementation details to the user:
+        * Bits of the runtime and intermediate libraries whose source code is likely not of interest to the user (but it might be);
+        * Intermediate frames from the stdlib;
+        * `::{closure}` symbols on async functions and blocks (even though they don't appear to be closures to the user);
+        * calls to `poll`.
+* **What are the sources for this story?**
+    * [Sergey Galich](https://github.com/rust-lang/wg-async-foundations/issues/69#issuecomment-803208049) reported this problem, among many others.
+* **Why did you choose Barbara to tell this story?**
+    * She knows about the desugarings that give rise to symbols like `::{closure}`, but she still finds them annoying to deal with in practice.
+* **How would this story have played out differently for the other characters?**
+    * Other characters might have wasted a lot of time trying to read through the stack trace in place before editing it.
+    * They might not have known how to trim down the stack trace to something that focused on their code, or it might have taken them much longer to do so.
+* **How does this compare to other languages?**
+    * Rust's async model does have some advantages, because the complete stack trace is available unless there is an intermediate `spawn`.
+    * Other languages have developed special tools to connect async functions to their callers, however, which gives them a nice experience. For example, Chrome has a [UI for enabling stacktraces that cross await points](https://www.html5rocks.com/en/tutorials/developertools/async-call-stack/#toc-enable).
+* **Why doesn't Barbara view this in a debugger?**
+    * Because it came in an issue report (or, freqently, as a crash report or email).
+* **Doesn't Rust have backtrace trimming support?**
+    * Yes, this **is** the reduced backtrace. You don't even want to know what the [full one](https://gist.github.com/eminence/0b3e697b7c4e686451ff0d37c169c89d) looks like. Don't click it. Don't!
```

Comment:
I plan to make this a link to this picture =)

![You clicked the link](https://user-images.githubusercontent.com/155238/112774854-fa651a00-9008-11eb-91ba-d093deb00b45.jpg)

Reply:
hit F5 on that gist URL 😄

Reply:
Ha! Genius.

Reply:
LOVE IT!